public/js/reader.js

```diff
@@ -29,14 +29,14 @@ const readerComponent = () => {
 						return {
 							id: i + 1,
 							url: `${base_url}api/page/${tid}/${eid}/${i+1}`,
-							width: d.width,
-							height: d.height,
+							width: d.width == 0 ? "100%" : d.width,
+							height: d.height == 0 ? "100%" : d.height,
 						};
 					});
 
-					const avgRatio = this.items.reduce((acc, cur) => {
+					const avgRatio = dimensions.reduce((acc, cur) => {
 						return acc + cur.height / cur.width
-					}, 0) / this.items.length;
+					}, 0) / dimensions.length;
```

Comment:
Why are we changing `this.items` to `dimentions`? Is there a particular reason behind it?

Reply:
Due to the 100% workaround, you otherwise end up with ``acc + "100%" / "100%"`` which probably ends up in NaN or something like that. Filtering these out it could end up /0 or some other funky issues. This way it just uses the unprocessed entries, so worst-case it's ``0/1`` (still wrong, but not an error).

Reply:
Oh you are right good point!